### PR TITLE
Feat/chart granularity

### PR DESCRIPTION
**Pull request by mmfallacy**
This PR implements granularity for chart Display component.

### CODE REVIEW

File: src/components/Chart/Display.svelte [hunk 1]
```diff
@@ -3,12 +3,14 @@
     import Chart from 'chart.js/auto';
     import ChartStreaming from 'chartjs-plugin-streaming';
     import ChartZoom from 'chartjs-plugin-zoom';
-    import { onMount } from 'svelte';
+    import { onDestroy, onMount } from 'svelte';
```
Comment (BastiDood) [inline src/components/Chart/Display.svelte:6]: This is apparently unused.
```suggestion
    import { onMount } from 'svelte';
```

File: src/components/Chart/Display.svelte [hunk 2]
```diff
@@ -17,6 +19,12 @@
 
     let canvas: HTMLCanvasElement;
 
+    let chart: Chart;
+
+    // Granularity is in the form of seconds. If left undefined,
+    // backend uses the predetermined quantum as the selected granularity
+    export let granularity: Granularity | undefined = undefined;
```
Comment (BastiDood) [inline src/components/Chart/Display.svelte:26]: Assigning `undefined` is redundant and contradictory.
```suggestion
    export let granularity: Granularity | undefined;
```

File: src/routes/Dashboard/Page.svelte [hunk 3]
```diff
@@ -11,14 +11,34 @@
...
+    import { assert } from '../../assert';
 
-    let value: Option = OPTIONS[0] || '';
+    let mac = 'ad:ad:ad:ad';
+
+    // Flatten objects to array of labels.
+    let GRANULARITY_OPTIONS = GRAN_OPTS.map(option => option.label);
+
+    $: console.log(GRAN_OPTS);
+    $: console.log(granularity);
```
Comment (BastiDood) [inline src/routes/Dashboard/Page.svelte:28]: Did we mean to leave this in?
```suggestion
```

File: src/routes/Dashboard/constants.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import COLORS from 'tailwindcss/colors';
 import type { Option } from '../../components/Select/types.ts';
```
Comment (BastiDood) [inline src/routes/Dashboard/constants.ts:2]: This is apparently unused.
```suggestion
```

**Review by BastiDood — CHANGES_REQUESTED** [review]

File: src/components/Chart/Display.svelte (hunk 2 above)
Comment (mmfallacy) [reply src/components/Chart/Display.svelte:26]: Not putting a default prop value of `undefined` raises a diagnostic that prop is possibly undefined on usage although typed as `T | undefined`.

Comment (mmfallacy) [reply src/components/Chart/Display.svelte]: [3762139](https://github.com/drippy-iot/dashboard/pull/20/commits/3762139755e7aa38bfc28efb525b4d5d7ae31819) adds `Granularity.REALTIME` as its default value instead.

**Review by BastiDood — APPROVED** [review]

**Pull request merged**